internal/karma/karma.conf.js

```diff
@@ -1,5 +1,56 @@
 // Karma configuration
 // GENERATED BY Bazel
+const path = require('path');
+const fs = require('fs');
+const tmp = require('tmp');
+
+// When karma is configured to use Chrome it will look for a CHROME_BIN
+// environment variable. This line points Karma to use puppeteer instead.
+// See
+// https://github.com/karma-runner/karma-chrome-launcher/blob/master/README.md#headless-chromium-with-puppeteer
+process.env.CHROME_BIN = require('puppeteer').executablePath();
+
+let files = [
+  TMPL_files
+];
+
+const manifestFile = path.join(process.env.TEST_SRCDIR, 'MANIFEST');
+if (fs.existsSync(manifestFile)) {
+  const manifest = {};
+  for (l of fs.readFileSync(manifestFile, 'utf8').split('\n')) {
+    const m = l.split(' ');
+    manifest[m[0]] = m[1];
+  }
+  const manifestKeys = Object.keys(manifest);
+  files = files.map((f) => {
+    if (manifestKeys.includes(f)) {
```

Comment:
this is O(n^2), but should be easy to avoid using a map

Reply:
Thanks. Updated to remove the double lookup. It looks like O(n*m) to me where n is the number of test files and m is the number of files in the MANIFEST.